python/ql/src/semmle/python/web/django/General.qll

```diff
@@ -0,0 +1,41 @@
+import python
+import semmle.python.regex
+import semmle.python.web.Http
+
+predicate django_route(CallNode call, ControlFlowNode regex, FunctionValue view) {
+    exists(FunctionValue url |
+        Value::named("django.conf.urls.url") = url and
+        url.getArgumentForCall(call, 0) = regex and
+        url.getArgumentForCall(call, 1).pointsTo(view)
+    )
+}
+
+/** As RegexStrings is an abstract class, we need to extend it to include the regex's used in django_routes */
```

Comment:
This comment is a bit curious to me. Surely the need to extend `RegexString` (no `s`) has nothing to do with the fact that it's an abstract class, but rather that we want to identify these strings as containing regexes, so that we can use the associated library support for regexes.

Also `regex's` should be `regexes`, in my opinion.

Reply:
This comment just needs to be cleaned up (removed). It was written to help me understand how abstract classes work.